src/components/ErrorBoundary.tsx

```diff
@@ -0,0 +1,96 @@
+import { Button, Stack, Typography } from '@mui/material';
+import {
+  Component,
+  ComponentType,
+  PropsWithChildren,
+  PropsWithRef,
+  ReactNode
+} from 'react';
+import {
+  //NavigateProps,
+  NavigateFunction,
+  Params,
+  useLocation,
+  useNavigate,
+  useParams,
+  Location
+} from 'react-router';
+import { AppHistory, useAppHistory } from '../hooks/useAppHistory';
+
+interface RouterProps {
+  navigate: NavigateFunction;
+  readonly params: Params<string>;
+  location: Location;
+}
+
+type WithRouterProps<T> = T & RouterProps;
+type OmitRouter<T> = Omit<T, keyof RouterProps>;
+
+function withRouter<T>(Component: ComponentType<OmitRouter<T> & RouterProps>) {
+  return (props: OmitRouter<T>) => {
+    const location = useLocation();
+    const navigate = useNavigate();
+    const params = useParams();
+    const appHistory = useAppHistory();
+    return (
+      <Component
+        location={location}
+        navigate={navigate}
+        params={params}
+        appHistory={appHistory}
+        {...props}
+      />
+    );
+  };
+}
+
+type ErrorBoundaryState = {
+  error: Error | null;
+};
+
+type ErrorBoundaryProps = {
+  message?: string;
+  children?: ReactNode;
+  appHistory?: AppHistory;
+};
+
+class ErrorBoundaryComponent extends Component<
+  WithRouterProps<PropsWithRef<PropsWithChildren<ErrorBoundaryProps>>>,
+  ErrorBoundaryState
+> {
+  constructor(props) {
+    super(props);
+    this.state = { error: null };
+  }
+
+  componentDidCatch(error) {
```

Comment:
Is this only for errors that are not being catched?

what happens to errors that are throw and catched and displayed as UI on the website?

Reply:
All caught errors will not bubble through. Only uncaught errors that will otherwise break the app with a white screen are caught here. If ran and you search for offers in Taipei, you'll see it working

Reply:
nice. Maybe an idea to wrap it around the whole app?

Reply:
Yea, that would've been ideal but we will no longer have access to the router. 
I placed the error boundary as close as possible to the router as I could.
And that is the `MainLayout` component which is common to all pages. 
I couldn't find any other common ancestor for all routes